homework-assignments-1/task4/task4.ts

```diff
@@ -0,0 +1,169 @@
+class Product {
+    constructor(
+        public name: string,
```

Comment:
Never ever expose class member properties to public unless you use them in derived classes.